packages/ui/src/SelectField.tsx

```diff
@@ -184,6 +242,35 @@ export const SelectField = <V,>({
     }
   }, [menuPortalTarget])
 
+  const onChangeMultipleFn = React.useCallback(
+    (values: SelectFieldOption<V>[] | null) => {
+      ;(onChange as (newValue: V[] | null) => void)(values === null ? null : values.map(({ value }) => value))
+    },
+    [onChange],
+  )
+
+  const onChangeFn = React.useCallback(
+    (option: SelectFieldOption<V | null>) => {
+      ;(onChange as (newValue: V | null) => void)(option === null ? null : option.value)
+    },
+    [onChange],
+  )
+
+  const onCreateOption = React.useCallback(
```

Comment:
Why do we need this one? AFAIU, react-select handles this logic automatically. I tried to remove it locally and it just worked. Then we'll also be able to remove `isCreatableGuard
`. Moreover, with the current implementation, a single-value creatable select does not work.

Reply:
Good catch @aigoncharov